src/Foundatio/Queues/QueueBase.cs

```diff
@@ -82,31 +86,43 @@ public virtual Task<IQueueEntry<T>> DequeueAsync(TimeSpan? timeout = null)
         public AsyncEvent<EnqueuedEventArgs<T>> Enqueued { get; } = new AsyncEvent<EnqueuedEventArgs<T>>(true);
 
         protected virtual Task OnEnqueuedAsync(IQueueEntry<T> entry) {
+            if (Enqueued == null)
```

Comment:
The intention of these additional null checks is to save the overhead of creating the `EnqueuedEventArgs`? 

When written like it is currently, both the additional check and the null propagation operator are required for thread safety, but it could be rewritten as below?
```csharp
var enqueued = Enqueued;
if (enqueued == null) {
    return Task.CompletedTask;
}

var args = new EnqueuedEventArgs<T> { Queue = this, Entry = entry };
return enqueued.InvokeAsync(this, args);
```